% !TeX root = ../../thesis.tex
\chapter{Objectives}
\label{ch:objectives}

\section{Hepatitis B Virus}

In a first part of this thesis, I am interested in obtaining a better understanding of the spatiotemporal dynamics that have contributed to the current geographic diversity of \gls{hbv} genotypes A, D, and E.
Unfortunately, one of the difficulties of analyzing \gls{hbv} genomic datasets is that the virus generally shows very little temporal signal.
To remedy this we perform temporal phylogeographic analysis of contemporary \gls{hbv} genomes and explore how and if the addition of several ``ancient'' \gls{hbv} genomes isolated from the tissue of mummies \cite{muhlemann2018ancient, ross2018paradox} render these data sets suitable for Bayesian phylogeographic inference without the need for external calibration information.
The genomes range between 500--4,500 years in age, and cover a geographic range from central Asia to the Korean peninsula; this provides more temporal signal to our data than has existed in previous studies, and helps us generate more accurate reconstructions of the temporal aspects of the evolutionary history of \gls{hbv}.

Beyond the accurate temporal inference of the evolutionary history of \gls{hbv}, I aim to reconstruct the migration history of the virus over evolutionary time.
Beyond the temporal signal that they contribute, the ancient genomes give us some information about the historical locations of different \gls{hbv} genotypes.
I aim to use these in tandem with our contempory dataset of genomes to infer when major migration events have taken place over the history of each \gls{hbv} genotype.
For this analysis, I strive to migigate any potential sampling bias that is introduced by over- or under-sampling of specific locations in our dataset.

Finally, I aim to phylogenetically place the novel set of \gls{hbv} genomes that are used in these analyses.
In doing so, I seek to determing a ``subgenotype'' label for each new sequence. %GB: can you provide some more information on how this will actually be done (from the e-mails/papers from Bobby)?

\section{Lassa Virus}
In the second part of this thesis, I seek to better understand the phylodynamics of \gls{lasv} in ``real time'' during seasonal outbreaks of Lassa fever in Nigeria.
A major hinderance to this process is the significant amount of ``preprocessing'' currently required to begin any Bayesian phylogenetic analysis pipeline.
It is a non-trivial process to add new data---even a single new genome---to an ongoing analysis; this often requires hand-made, bespoke preprocessing workflows.
Here, I present a new method for automating one such work flow to facilitate rapid analysis of \gls{lasv} genomes during a seasonal outbreak.
This workflow removes much of the manual labor required to being Bayesian phylogenetic analysis, and represents the first step toward creating a complete online \gls{beast} ecosystem.

\section{Online BEAST roadmap}

Finally, we present a framework by which this process may be extended in the future.
Specifically, we propose a methodology by which our pipeline may interact directly with existing genomic database tools to provide a turnkey database-to-analysis pipeline that is easily used and modified by any user, regardless of their familiarity with the existing phylogenetic analysis ecosystem.
We further describe some of the future work necessary to make this system robust to more phylogenetic models, and able to be run from start to finish without user intervention on cloud compute resources so that they are accessible to all researchers, regardless of access to physical compute resources.
These improvements together constitute an ecosystem in which full online Bayesian phylogenetic analyses may be performed without exposing the user to the back-end operations, increasing both usability and interpretability.


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Keep the following \cleardoublepage at the end of this file,
% otherwise \includeonly includes empty pages.
\cleardoublepage

% vim: tw=70 nocindent expandtab foldmethod=marker foldmarker={{{}{,}{}}}
